go/types/list.go

```diff
@@ -99,6 +117,7 @@ func (l List) Get(idx uint64) Value {
 type MapFunc func(v Value, index uint64) interface{}
 
 func (l List) Map(mf MapFunc) []interface{} {
+	// TODO: This is bad API. It should have returned another List.
```

Comment:
So, not documenting it because it's not a good API and we want to change it at some point? That... makes sense to me to some extent, but IMO it would be better to just say that in the godoc with a link to the issue (assuming an issue exists).